### PR TITLE
[Text Generation][V2] NonKVCachePipeline 

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Description

Added the `TestGenerationPipelineNoKVCache`. This pipeline processes the prompt and returns the new token. That's it.
Its main functionality is mapping prompt tokens to logits, instrumental for computing the perplexity of the model given a dataset

### Testing
Updated the integration tests to cover the case of non-kv-cache inference.

### Example Use

```python
from deepsparse.v2.text_generation import TextGenerationPipelineNoCache

prompt = ["Some funny prompt", "Why are you so"]

pipeline = TextGenerationPipelineNoCache(model_path="hf:mgoin/TinyStories-1M-ds",
                                         onnx_model_name="model-orig.onnx",
                                         sequence_length=20)

out = pipeline(prompt=prompt,
               include_prompt_logits=True,
               generation_kwargs=dict(output_scores=True))

for gen in out.generations:
    print(gen)
```

```bash
text='.' score=array([[ 2.9344807 , -0.03345669, -4.11256   , ..., -6.9316325 ,
        -4.6005425 ,  1.1827914 ],
       [ 7.008805  , -0.11603884, -7.1837015 , ..., -7.0405912 ,
        -2.386351  , -2.2007818 ],
       [ 6.348213  , -2.2960157 , -6.433192  , ..., -6.5930486 ,
        -5.8315077 , -0.58804405],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ]], dtype=float32) finished=True finished_reason='length' # notice that logits get zero padding from the end, this is because all logits need to have the same shape (the length of the longest prompt in the input +1)
text=' sad' score=array([[ 2.560934 ,  1.1993233, -6.670935 , ..., -7.3002615, -3.823823 ,
         1.8125833],
       [-1.1050931, -2.4256568, -7.3015127, ..., -6.1500154, -4.074909 ,
         1.8155754],
       [ 6.172593 , -2.2252593, -9.146653 , ..., -7.70834  , -4.810748 ,
         0.3985293],
       [ 1.4988875,  1.0973434, -4.4714937, ..., -4.8026247, -1.1791464,
         1.6924176]], dtype=float32) finished=True finished_reason='length'
```

### Next steps
- Create a parent`TextGenerationPipeline` operator that can either choose to use the kv-cache or non-kv cache version of the pipeline, depending on the topology of the ONNX model
- Move the overwriting of the transformer inputs to some high-level function
- Use the V2 pipeline for Perplexity calculation
- swap GraphRouter for LinearRouter in `TextGenerationPipelineNoKVCache`

### CODE REVIEW

**Review by bfineran — APPROVED** [review]

File: src/deepsparse/transformers/helpers.py [hunk 1]
```diff
@@ -30,11 +30,7 @@
 from onnx import ModelProto
 
 from deepsparse.log import get_main_logger
-from deepsparse.utils.onnx import (
-    _MODEL_DIR_ONNX_NAME,
-    model_to_path,
-    truncate_onnx_model,
-)
+from deepsparse.utils.onnx import MODEL_ONNX_NAME, model_to_path, truncate_onnx_model
```
Comment (dsikka) [inline src/deepsparse/transformers/helpers.py:33]: can we double check the changes to `MODEL_ONNX_NAME` and `get_deployment_path` wont break anything in v1 or the v2 kv cache pathway?

Comment (dbogunowicz) [reply src/deepsparse/transformers/helpers.py:33]: What do you mean exactly? All the tests pass, so the changes should not influence the existing code. What I mean by that is that we only add functionality, not change any existing functionality.

File: src/deepsparse/transformers/pipelines/text_generation/prep_for_generation.py [hunk 2]
```diff
@@ -90,9 +94,13 @@ def run(
...
             "token_generator": token_generator,
         }
+
         output = {
             "tokens": token_generator.tokens,
             "kv_cache": kv_cache,
             "in_generation": True,
         }
+        if kv_cache is None:
+            output = NLEngineOutputs(**output, engine_outputs=prompt_logits)
```
[the remark this answers is not in the document]
Comment (dbogunowicz) [reply src/deepsparse/transformers/pipelines/text_generation/prep_for_generation.py:104]: Well, this was the most elegant how to "bypass" the irrelevant operators (that would normally be there for `TextGenerationPipeline`) between `prepare_generation` and  `generate_new_token`. 
I need to call the `prep_for_generation` first to initialize the `token_generator`.
Then I need to `generate_new_token`, that expects `NLEngineOutputs` input. 

I am aware that this is a bit awkward, but this was the most elegant and least "code intensive" solution that I could come up with.

File: src/deepsparse/transformers/pipelines/text_generation/pipeline_no_kv_cache.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        )
+        generate_new_token = GenerateNewTokenOperator(
+            tokenizer=self.tokenizer, force_max_tokens=True
+        )
+        compile_generations = CompileGenerations()
+        join_output = JoinOutput(tokenizer=self.tokenizer)
+        process_outputs = ProcessOutputs(tokenizer=self.tokenizer)
+
+        ops = {
+            "process_input": process_inputs,
```
[the remark this answers is not in the document]
Comment (dsikka) [reply src/deepsparse/transformers/pipelines/text_generation/pipeline_no_kv_cache.py:93]: @dbogunowicz

File: src/deepsparse/transformers/pipelines/text_generation/prep_for_generation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dsikka) [reply src/deepsparse/transformers/pipelines/text_generation/prep_for_generation.py:104]: There is no output schema set for `generate_new_token` - you can make the typing in its run function a Union between whatever schema you want to use for non-kv cache and `NLEngineOutputs` and have whatever inputs you need. Also, if all you need is the TokenGenerator from `prep_for_generation`, that's a separate operator you can use independently of `prep_for_generation`

File: src/deepsparse/transformers/pipelines/text_generation/pipeline_no_kv_cache.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            "engine_operator": engine_operator,
+            "prepare_generation": prepare_generation,
+            "generate_new_token": generate_new_token,
+            "compile_generations": compile_generations,
+            "join_output": join_output,
+            "process_outputs": process_outputs,
+        }
+        routes = {
+            "process_input": "SPLIT",
```
Comment (dbogunowicz) [inline src/deepsparse/transformers/pipelines/text_generation/pipeline_no_kv_cache.py:102]:
```suggestion
            "parse_inputs": "process_input",
            "process_input": "SPLIT",
```

**Review by bfineran — APPROVED** [review]

**Review by dsikka — APPROVED** [review]
Thank you for the changes - this looks great.
I think the only thing that we're missing is the logic check which runs the appropriate pipeline by checking if kv_cache is present. We should do this in a follow-up PR.
